### ISSUE TITLE
Key4hep won't pull in python 3.10

### ISSUE THREAD

**Issue by joequant** [key4hep/key4hep-spack#624]
!!!IMPORTANT!!!: To facilitate faster and easier response to your issue please provide in addition to the description of the issue also the following information

 - OS version: Mageia Linux
 - Compiler version:  GCC 14.0
 - Package version: [1cd6239](https://github.com/key4hep/key4hep-spack/commit/1cd6239bc224630205d0cec2723afb03fcc13b14)
 - Reproduced by: exact steps to reproduce the problem: checkout, setup environment (ROOT versions, LCG release), cmake, build, run...

spack spec key4hep-stack

The preferred version of python is 3.11.9 but when I try to build key4hep-stack it pulls in python 3.10.14 

Trying to figure out why, and ideas for debugging.

**Comment by joequant** [comment]
Its a dependency on py-protobuf+cpp

**Comment by jmcarcell** [comment]
Yeah I checked once and remember reaching the same conclusion, it was impossible to have python 3.11 having a single version of protobuf. One possible option would be to have `py-torch` install its own protobuf, it's even commented here: https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/py-torch/package.py#L106-L108, but I haven't tried this yet.
